### ISSUE TITLE
Connection errors should "end" readable streams

### ISSUE THREAD

**Issue by marshall007** [neumino/rethinkdbdash#306]
Given that there is (at least currently) no way to recover when a connection is dropped on a cursor, I think we should close any readable streams bound to that cursor. For example:

``` js
const stream = r.table('test')
.toStream({ readable: true })
.pipe(/* transform */)
.pipe(/* writable */)

stream.on('error', ...) // errors on readable stream remain uncaught
stream.on('end', ...)
```

The way streams work in Node, an `error` event on the readable stream does not propagate through the other streams. You can work around this by attaching event listeners before you `.pipe()`, but since you can't recover from these errors anyway, I think we should also emit an `end` even, which definitely gets propagated.

**Comment by neumino** [comment]
Sounds pretty reasonable for me.  Fixed in 2.3.24.

**Comment by marshall007** [comment]
Thanks @neumino!
